ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -66,21 +65,21 @@ class ResourceManager implements Closeable, IngestionResourceManager {
     private String identityToken;
     private IngestionResourceSet ingestionResourceSet;
 
-    public ResourceManager(Client client, long defaultRefreshTime, long refreshTimeOnFailure, @Nullable CloseableHttpClient httpClient) {
+    public ResourceManager(Client client, long defaultRefreshTime, long refreshTimeOnFailure, @Nullable HttpClient httpClient) {
         this.defaultRefreshTime = defaultRefreshTime;
         this.refreshTimeOnFailure = refreshTimeOnFailure;
         this.client = client;
         timer = new Timer(true);
         // Using ctor with client so that the dependency is used
         this.httpClient = httpClient == null
-                ? new NettyAsyncHttpClientBuilder().responseTimeout(Duration.ofMinutes(UPLOAD_TIMEOUT_MINUTES)).build()
-                : new HttpClientWrapper(httpClient);
+                ? new NettyAsyncHttpClientProvider().createInstance(new HttpClientOptions().setResponseTimeout(Duration.ofDays(UPLOAD_TIMEOUT_MINUTES)))
```

Comment:
Instead of this, why not the default provider, or using the methods from data?

Reply:
I agree. I think we should avoid using specific providers in code and use default since default should fall back to the consumer's choice if Netty is explicitly excluded in the consumer's pom file.

Reply:
[Removed here. Also set timeout to 10 minutes, not 10 days. Not sure if that was intentional or not.](https://github.com/Azure/azure-kusto-java/pull/335/commits/286bbe10c5026acb825b609791de00ea42404f93)